tests/backend/test_compatibility.py

```diff
@@ -66,3 +77,58 @@ class OlderHandshakeChallengeSchema(BaseSchema):
     # Backend API < 2.4 with newer clients
     data = older_handshake_challenge_serializer.dumps(old_data)
     assert handshake_challenge_serializer.loads(data) == compat_data
+
+
+# This test would be useless when all clients and server will be up to date
```

Comment:
As mentioned above, I would keep this test for older client/newer backend compatibility.